PyTDF/TDataFrame.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import print_function
+from .Node import Node
+
+class TDataFrame(Node):
+    
+    """
+    The Python equivalent
+    of ROOT C++'s TDataFrame 
+    data structure
+
+    """
+
+    def __init__(self, treename, filelist, npartitions=None):
+        # TODO (shravan97) : Implement different types of constructors
+
+        super(TDataFrame, self).__init__(None, None)
+        self.filelist = filelist
+        self.treename = treename
+        self.npartitions = npartitions
+        self.chain = ROOT.TChain(treename)
+        self._tdf = ROOT.Experimental.TDataFrame(self.chain)
```

Comment:
I understand the creation of the real TDataFrame should be moved to the code that calls the mapper? @diocas @bluehood 
In the case of distributed execution, it is DistROOT that constructs the real TDataFrame from a range. If it is a local execution, it will be the code that actually calls the mapper locally.

Reply:
@etejedor Yup certainly. Since I'm yet to commit the mapper, I've added the creation of real TDF object here temporarily. This will be removed when I commit the mapper. And yes, I agree with you on the fact that it'll be created by DistROOT when we go distributed

Reply:
yes, the pyTDF should not have to build a TDF.

as discussed the mapper function takes a TDF object as argument, replays the booking of transformations and actions, and returns the list of results.

as such the mapper is general enough that it can be passed to a number of different "executors": a distributed executor (`distROOT`) first and foremost, but also a local executor for testing purposes, etc.